CarRentals/CarRentals/Controllers/Crud.cs

```diff
@@ -0,0 +1,50 @@
+﻿using CarRentals.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CarRentals.Controllers
+{
+    public abstract class Crud<T> : IDataProcessing<T> where T : IModelForCrud
```

Comment:
IMHO this class became somehow dummy. I recommend merging this code with `ManageJson` class. Make this class a JSON implementation of `IDataProcessing`... like `JsonStorage`. In that way if you later decide to use some other sort of storage, you can add something like `XmlStorage`.

Reply:
I don't know if I understood well. you say put the code from ManajeJson here? 
IDataProcessing is the interface which indicates methods Create, Delete, Update and Get and no more. I would have to create a new interface with Json methods (SaveChanges, ReadJson, etc)? and it will implemented by Crud?

Reply:
More or less... yes.

- `IDataProcessing` stays the same. No modifications.
- No need to create a new interface.
- `JsonStorage` will implement `IDataProcessing` and your `ReadFromJson` and `SaveToJson` will be private methods. Have in mind that those are _**implementation details**_ of your JSON solution for the storage. I believe that's the way to make your code simpler. Remember that the consumer of your CRUD operations doesn't care how you store it to json files. That's your `JsonStorage`'s responsibility.

Side note: your current code isn't wrong. But I think you prematurely added a complexity layer that's not needed. As programmers we usually tend to do such things but we should avoid it. It makes everything harder, slower, inefficient. Only add that extra layer when it's needed.

### An alternative

You can continue working with your json handling solution as a separate class... but if you continue that way try with DI (dependency injection).

Reply:
@matiasbeckerle I have made all the changes you told me. Also I have modularized the Update method to avoid duplicate code. I think this code is correct but if I should have to change somethig else please tell me.